### DISCUSSION TITLE
How to always get a line space between functions and similar?

### DISCUSSION THREAD

**Discussion by alanwilter** [rust-lang/rustfmt#5992] [Q&A]
I'm new to `rust` and I use `vscode`. I found it nice it can keep the code mostly properly formatted.

For example, if I put two space lines between the functions below, when saving, this is immediately rectified.

```rust
fn foo() {
    println!("Hello");
}

fn main() {
    println!("Hello Main!");
}
```

However, if remove the space line, the format accepts that but I don't like this!

```rust
...
}
fn main() {
...
```
How can I enforce always a space (one line space) between functions and similar?

**Comment by ytmimi** [comment]
Maybe `blank_lines_lower_bound` is what you're looking for, though it'll affect more than just the space between top level items.

**Note `blank_lines_lower_bound` is a nightly only configuration option**.

Running with `blank_lines_lower_bound=1` on the following input:
```rust
fn foo() {
    println!("Hello");
}
fn main() {
    println!("Hello Main!");
}
```

Produces this output:
```rust
fn foo() {

    println!("Hello");
}

fn main() {

    println!("Hello Main!");
}
```

**Reply by alanwilter** [reply]
Thanks @ytmimi but where do I set this option? Like I said, I'm using `vscode`. And which `nightly`: `rustup` or Vscode rust-analyser extension?

**Reply by ytmimi** [reply]
[rustfmt is configured via a `rustfmt.toml` file](https://github.com/rust-lang/rustfmt?tab=readme-ov-file#configuring-rustfmt).
